### PR TITLE
Refacto EntryPoint and Authenticator

### PR DESCRIPTION
**Pull request by amenophis**
Hi @Nyholm,
Little refacto to simplify the bundle code and use more the SDK.

### CODE REVIEW

File: src/Security/Auth0EntryPoint.php [hunk 1]
```diff
@@ -15,43 +14,28 @@
...
-            'redirect_uri' => $this->httpUtils->generateUri($request, $this->loginCheckRoute),
-            'state' => $csrfToken->getValue(),
-            'scope' => $this->configuration->buildScopeString(),
-            // https://auth0.com/docs/universal-login/i18n
-            'ui_locales' => $request->getLocale(),
-        ];
-
-        return new RedirectResponse(sprintf('https://%s/authorize?%s', $this->loginDomain ?? $this->configuration->getDomain(), http_build_query($query)));
+        return new RedirectResponse(
+            $this->auth0->signup(
```
Comment (Nyholm) [inline src/Security/Auth0EntryPoint.php:34]: Isnt `login()` more appropriate?

File: src/Security/Authentication/Auth0Authenticator.php [hunk 2]
```diff
@@ -57,53 +47,28 @@ public static function getSubscribedServices()
...
+                };
+            }
 
-        $userProviderCallback = null;
-        if (null !== $up = $this->get(Auth0UserProviderInterface::class)) {
-            $userProviderCallback = static function () use ($up, $userModel) {
-                return $up->loadByUserModel($userModel);
-            };
+            return new SelfValidatingPassport(new UserBadge($userModel->getEmail(), $userProviderCallback), [new Auth0Badge($userModel)]);
+        } catch (Auth0Exception $e) {
```
Comment (Nyholm) [inline src/Security/Authentication/Auth0Authenticator.php:69]: We can move this up so the `catch`-block is as small as possible

File: src/Security/Authentication/Auth0Authenticator.php [hunk 3]
```diff
@@ -57,53 +47,28 @@ public static function getSubscribedServices()
...
+                    return $up->loadByUserModel($userModel);
+                };
+            }
 
-        $userProviderCallback = null;
-        if (null !== $up = $this->get(Auth0UserProviderInterface::class)) {
-            $userProviderCallback = static function () use ($up, $userModel) {
-                return $up->loadByUserModel($userModel);
-            };
+            return new SelfValidatingPassport(new UserBadge($userModel->getEmail(), $userProviderCallback), [new Auth0Badge($userModel)]);
```
Comment (Nyholm) [inline src/Security/Authentication/Auth0Authenticator.php:68]: We cannot be sure that we have `email`. 

Let's use `user_id` for now and #45 will solve this later.

**Review by Nyholm — COMMENTED** [review]
Thank you!

If you have not passed a TransientStorage to the SDK then this authentication will fail. That is fine. It is sub-optimal DX but it is more transparent with the SDK.

We need to make sure we give the user working config example in the docs.

**Comment by amenophis** [comment]
I have not configured any transientStorage in the configuration, and it works out of the box

File: src/Security/Auth0EntryPoint.php (hunk 1 above)
Comment (amenophis) [reply src/Security/Auth0EntryPoint.php:34]: sure !

**Comment by Nyholm** [comment]
Oh. I see that it is using `CookieStorage` as default: https://github.com/auth0/auth0-PHP/blob/main/src/Configuration/SdkConfiguration.php#L396

Good good then. As long as the config example in the readme is working. We can improve the docs later with PSR-6 stuff if we want to.

**Comment by Nyholm** [comment]
Could you please rebase so the CI is green?

**Comment by amenophis** [comment]
The 8.0.0-BETA2 tag doesn't exists, so composer fail to install deps

**Comment by Nyholm** [comment]
Yes. I was wrong before. 

PR https://github.com/Happyr/auth0-bundle/pull/48 is merged now. Master is green https://github.com/Happyr/auth0-bundle/actions/runs/1106719094

**Comment by amenophis** [comment]
Ok, will rebase tomorrow !

**Comment by amenophis** [comment]
Done

**Review by Nyholm — APPROVED** [review]
Thank you.

I need to confirm that custom login domains work.

**Pull request merged**